runhouse/cli_utils.py

```diff
@@ -0,0 +1,522 @@
+                        func_start_time_utc = active_function_calls[0].get(
+                            "start_time", None
+                        )
+
+                        # casting func_start_time_utc to datetime format
+                        func_start_time_utc = datetime.fromtimestamp(
+                            func_start_time_utc, tz=timezone.utc
+                        )
+
+                        # func_end_time_utc = current time. Making sure it is in the same format as func_start_time_utc,
+                        # so we could calculate function's running time.
+                        func_end_time_utc = datetime.fromtimestamp(
+                            time.time(), tz=timezone.utc
+                        )
+
+                        func_running_time = (
+                            func_end_time_utc - func_start_time_utc
+                        ).total_seconds()
+
+                        is_func_running: str = f" [italic bright_green]Running for {func_running_time} seconds[/italic bright_green]"
+
+                    elif (
+                        resource_type == "runhouse.Function"
+                        and not active_function_calls
+                    ):
+                        is_func_running: str = " [italic bright_yellow]Currently not running[/italic bright_yellow]"
+
+                    else:
+                        is_func_running: str = ""
+
+                    resource_info_str = resource_info_str + is_func_running
+
+                    console.print(resource_info_str)
+
+
+def print_cloud_properties(cluster_config: dict):
+    from runhouse.main import console
+
+    cloud_properties = cluster_config.get("launched_properties", None)
+    if not cloud_properties:
+        return
+    cloud = cloud_properties.get("cloud")
+    instance_type = cloud_properties.get("instance_type")
+    region = cloud_properties.get("region")
+    cost_per_hour = cloud_properties.get("cost_per_hour")
+
+    has_cuda = cluster_config.get("has_cuda", False)
+    cost_emoji = "💰" if has_cuda else "💸"
+
+    num_of_cpus = cloud_properties.get("num_cpus") or len(cluster_config.get("ips"))
+    num_of_gpus = 0
+    cluster_accelerators = cloud_properties.get("accelerators", None)
+    gpu_types = set()
+    if cluster_accelerators:
+        for k, v in cluster_accelerators.items():
+            num_of_gpus = num_of_gpus + int(v)
+            gpu_types.add(k)
+
+    console.print(
+        f"[reset]🤖 {cloud} {instance_type} cluster | 🌍 {region} | {cost_emoji} ${cost_per_hour}/hr"
+    )
+    cpus_gpus_info_str = f"CPUs: {int(float(num_of_cpus))}"
+    if num_of_gpus > 0:
+        gpu_types_str = gpu_types.pop()
+        for gpu_type in gpu_types:
+            gpu_types_str = gpu_types_str + f", {gpu_type}"
+        cpus_gpus_info_str = (
+            cpus_gpus_info_str + f" | GPUs: {num_of_gpus} (Type(s): {gpu_types_str})"
+        )
+    console.print(f"[reset]{cpus_gpus_info_str}")
```

Comment:
do you have an example for what this looks like rendered in the CLI?

Reply:
![Screenshot 2024-09-30 at 9 31 32](https://github.com/user-attachments/assets/f0e5b446-2ab3-4fde-a4cc-a2d2ce3d7fef)